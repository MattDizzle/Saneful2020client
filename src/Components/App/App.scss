@import url('https://fonts.googleapis.com/css2?family=Press+Start+2P&display=swap');

.App{
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  height: 100vh;
  width: 100vw;
}

main{
  width: 100%;
  height: 100%;
}