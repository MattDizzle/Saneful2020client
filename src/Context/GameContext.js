import React, { Component } from 'react';
import SaveApiService from '../services/save-service';

const GameContext = React.createContext({

  gameData: {
    saved_game_id: 0,
    current_x_coord: 0,
    current_y_coord: 0,
    money_counter: 0,
    health_points: 0,
    health_points_max: 0,
    sanity_points: 0,
    sanity_points_max: 0,
    dead: false,
    character_skin: 1,
    elapsed_time: 0,
  },
  setGameData: () => { },
  newGame: () => { },
  loadGame: () => { },
  saveGame: () => { },
  currentGameExists: () => { },
});

export default GameContext;

export class GameProvider extends Component {
  constructor(props) {
    super(props);
    this.state = {
      gameData: {
        saved_game_id: 0,
        current_x_coord: 0,
        current_y_coord: 0,
        money_counter: 0,
        health_points: 0,
        health_points_max: 0,
        sanity_points: 0,
        sanity_points_max: 0,
        dead: false,
        character_skin: 1,
        elapsed_time: 0,
      }
    };
  }

  setGameData = (gameData) => {
    this.setState({
      gameData: {
        saved_game_id: gameData.saved_game_id,
        current_x_coord: gameData.current_x_coord,
        current_y_coord: gameData.current_y_coord,
        money_counter: gameData.money_counter,
        health_points: gameData.health_points,
        health_points_max: gameData.health_points_max,
        sanity_points: gameData.sanity_points,
        sanity_points_max: gameData.sanity_points_max,
        dead: gameData.dead,
        character_skin: gameData.character_skin,
        elapsed_time: gameData.elapsed_time,
      }
    });
  };

  newGame = async (newGameData = {
    current_x_coord: 0,
    current_y_coord: 2,
    money_counter: 0,
    health_points: 100,
    health_points_max: 100,
    sanity_points: 100,
    sanity_points_max: 100,
    dead: false,
    character_skin: 1, //this will be passed in by a selection by the user when they create a new game
    elapsed_time: 0
  }) => {
    let result;

    try {
      result = await SaveApiService.postNewGameData(newGameData);
    } catch (error) {
      console.error(error);
    }

    this.setGameData(result);
  };

  currentGameExists = async () => {
    let result;

    try {
      result = await SaveApiService.getSaveGameData();
      const currentGame = result.find(game => game.dead === false);

      if (currentGame) {
        return true;
      }

      return false;
    } catch (error) {
      console.error(error);
    }
  };

  loadGame = async () => {
    let result;

    try {
      result = await SaveApiService.getSaveGameData();
      const currentGame = result.find(game => game.dead === false);

      if (currentGame) {
        this.setGameData(currentGame);
        console.log('loading current game...', currentGame);
      }
      else {
        this.newGame();
        console.log('no current game');
      }

    } catch (error) {
      console.error(error);
    }

  };

  saveGame = async () => {
    try {
      await SaveApiService.patchGameData(this.state.gameData);
    } catch (error) {
      console.error(error);
    }
  };

  render() {
    const value = {
      gameData: this.state.gameData,
      setGameData: this.setGameData,
      newGame: this.newGame,
      loadGame: this.loadGame,
      saveGame: this.saveGame,
      currentGameExists: this.currentGameExists,
    };
    return (
      <GameContext.Provider value={value}>
        {this.props.children}
      </GameContext.Provider>
    );
  }
}