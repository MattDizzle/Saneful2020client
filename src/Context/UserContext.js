import React, { Component } from 'react';
import TokenService from '../services/token-service';

const UserContext = React.createContext({
  user: {},
  error: null,
  setError: () => { },
  clearError: () => { },
  setUser: () => { },
  processLogin: () => { },
  processLogout: () => { },
  setStartClicked: () => { },
});

const nullUser = {
  id: null,
  email: null,
  username: null,
};

export default UserContext;

export class UserProvider extends Component {
  constructor(props) {
    super(props);

    const state = {
      user: nullUser,
      error: null,
      startClicked: true,
    };

    const jwtPayload = TokenService.parseAuthToken();

    if (jwtPayload)
      state.user = {
        id: jwtPayload.user_id,
        email: jwtPayload.sub,
        username: jwtPayload.user_name,
      };

    this.state = state;
  }

  setError = error => {
    this.setState({ error });
  };

  setStartClicked = clicked => {
    this.setState({ startClicked: clicked });
  };

  clearError = () => {
    this.setState({ error: null });
  };

  setUser = user => {
    this.setState({ user });
  };

  processLogin = authToken => {
    TokenService.saveAuthToken(authToken);
    const jwtPayload = TokenService.parseAuthToken();
    //need to add username
    this.setUser({
      username: jwtPayload.user_name,
      id: jwtPayload.user_id,
      email: jwtPayload.sub
    });
  };

  processLogout = () => {
    TokenService.clearAuthToken();
    this.setUser({});
  };

  render() {
    const value = {
      user: this.state.user,
      error: this.state.error,
      startClicked: this.state.startClicked,
      setError: this.setError,
      clearError: this.clearError,
      setUser: this.setUser,
      processLogin: this.processLogin,
      processLogout: this.processLogout,
      setStartClicked: this.setStartClicked,
    };
    return (
      <UserContext.Provider value={value}>
        {this.props.children}
      </UserContext.Provider>
    );
  }

}