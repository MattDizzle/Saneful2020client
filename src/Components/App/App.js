import React, { useContext } from "react";
import { Switch } from "react-router-dom";
import LeaderBoard from "../../Routes/LeaderBoard";
import Login from "../../Routes/Login";
import MainGame from "../../Routes/MainGame";
import Dashboard from "../../Routes/Dashboard";
import Register from "../../Routes/Register";
import Header from "../../Components/Header";
import PrivateRoute from "../PrivateRoute/PrivateRoute";
import PublicRoute from "../PublicRoute/PublicRoute";
import UserContext from '../../Context/UserContext';

import "./App.scss";

const App = () => {
  const context = useContext(UserContext);
  return (
    <div className="App">
      {context.startClicked && <Header />}
      <main>
        <Switch>
          <PublicRoute exact path="/" component={Login} />
          <PublicRoute path="/register" component={Register} />
          <PrivateRoute path="/maingame" component={MainGame} />
          <PrivateRoute path="/dashboard" component={Dashboard} />
          <PrivateRoute path="/leaderBoard" component={LeaderBoard} />
        </Switch>
      </main>
    </div>
  );
};

export default App;
