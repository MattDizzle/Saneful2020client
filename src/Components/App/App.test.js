import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";
import { UserProvider } from '../../Context/UserContext';
import { GameProvider } from '../../Context/GameContext';
import App from "./App";

describe(`App Component`, () => {
    describe(`Smoke test`, () => {
        it("should render without crashing", () => {

            const div = document.createElement('div');

            ReactDOM.render(
            <BrowserRouter>
              <UserProvider>
                <GameProvider>
                  <App />
                </GameProvider>
              </UserProvider>
            </BrowserRouter>,
            div);

              ReactDOM.unmountComponentAtNode(div);
          });

    })

    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {

          const div = document.createElement('div');

            const tree = renderer
                .create(
                  <BrowserRouter>
                  <UserProvider>
                    <GameProvider>
                      <App />
                    </GameProvider>
                  </UserProvider>
                </BrowserRouter>)
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });

})


