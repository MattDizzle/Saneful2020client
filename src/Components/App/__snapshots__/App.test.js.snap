// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="App"
>
  <header
    className="Header"
  >
    <nav>
      <ul
        className="nav-bar"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
        <li>
          <a
            href="/register"
            onClick={[Function]}
          >
            Sign up
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <div
      className="Login"
    >
      <img
        alt="logo"
        className="logo"
        src="navywhitelogo.png"
      />
      <form
        onSubmit={[Function]}
      >
        <input
          name="email"
          onChange={[Function]}
          placeholder="Enter email..."
          type="email"
          value=""
        />
        <input
          name="password"
          onChange={[Function]}
          placeholder="Enter password..."
          type="password"
          value=""
        />
        <button
          className="loginButton"
          type="submit"
        >
          Log In
        </button>
        <div
          className="error-message"
          role="alert"
        />
        <div
          className="demo-creds"
        >
          <h4>
            guest account
          </h4>
          <p>
            email: guest@saneful.com
          </p>
          <p>
            password: San3fu!!
          </p>
        </div>
      </form>
    </div>
  </main>
</div>
`;
